cmake-modules/global_compile_options.cmake

```diff
@@ -12,7 +12,7 @@ if(MSVC)
   string(REGEX REPLACE "/W3" "/W4" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
   #https://stackoverflow.com/questions/37527946/warning-unreferenced-inline-function-has-been-removed
-  add_compile_options(/W4 ${WARNINGS_AS_ERRORS_FLAG} /wd5031 /wd4668 /wd4820 /wd4255 /wd4710 /analyze)
```

Comment:
why are we removing `/analyze`? We probably should't, as some users may have it on, and we need to compile clean.

Reply:
I am making some experiments.
Noticed that after removing this, compiling code from scratch takes less than 2 minutes, compared to more than 10 minutes when this is ON.
I think we should only use it as part of CI (enable it with some env var). This would speed up Windows development flow.

Reply:
We can add it to nightly builds then. I agree nothing critical has came out after using it, only ideally we don't want any warnings coming from it.

Reply:
I would be more than happy to move it to nightly only!  However, there is this `all times situation` where nightly build issues might like of an owner (or even an orchestrator to look for the root cause and delegate).
What happen, for example, in the future, if the SDK is super famous, and we get hundreds of PRs from the community, we merge hundreds of commits and then the build fails during the night with multiple static code analysis warnings. Will we ask the contributors to come back and fix?  Will we fix them now?  
So, better to catch it directly from public CI pipelines, I think.

Reply:
That's not a bad idea. Although I think having it in CI hit a nice balance between slowing down dev productivity and increasing the distance between a change and its failure. I largely want nightly build failures to be one-off outliers and remain green otherwise. You don't want recurring failures because of the /analyze warning flag and then fish out which of the N commits that day were the root cause of the problem.

The fact that CI takes 6 minutes vs 18, might be OK here since at a certain point there is diminishing returns of reducing CI time since the bottleneck at that point tends to be code reviews and dev work anyway. If we think it is still worth it to save that time, I am OK with starting it in the nightly build and see how many failures we get in a week/month, and move it back to CI if it is too much noise/overhead.